### PR TITLE
RMB-735: Reject unsupported sort modifiers in getWithOptimizedSql

### PR DESCRIPTION
**Pull request by julianladisch**
Sorting in CQL is expessed using the sortby keyword followed by
a list of sort-keys, each expressed as a CQL index data optionally
followed by one or more index modifiers — see
http://zing.z3950.org/cql/sorting.html#6.
For example, title/sort.descending/sort.ignorecase.

RMB handles the sort.ascending and sort.descending modifiers.
It rejects all other sort modifiers with a
CQLFeatureUnsupportedException:
https://github.com/folio-org/raml-module-builder/blob/v31.1.0/cql2pgjson/src/main/java/org/folio/cql2pgjson/model/CqlModifiers.java#L71

For optimized queries (that we use for instance.title only)
we have some lazy code that doesn't reject other sort modifiers:
https://github.com/folio-org/raml-module-builder/blob/v31.1.0/domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java#L1097

This pull request fixes this inconsistency: Reject unsupported sort
modifiers in optimized queries (PgUtil.getWithOptimizedSql) as well.

Example that demonstrates the inconsistency:

((keyword=water) and (source=marc)) sortby title/sort.missingLow/sort.ascending/sort.respectCase

is accepted; but if I add another sort-key, hrid/sort.descending
— which you will note does not include a "missing-value" modified — yielding

((keyword=water) and (source=marc)) sortby title/sort.missingLow/sort.ascending/sort.respectCase hrid/sort.descending

then that sort is rejected with "Unsupported modifier sort.missinglow".
That can't be right.

Note that RMB uses schema.json to determine whether the index is
case sensitive or accents sensitive. See https://issues.folio.org/browse/RMB-734
for the feature request to allow those sort modifiers that match
the index definition in schema.json.

### CODE REVIEW

**Review by MikeTaylor — APPROVED** [review]
Looks reasonable to me.

(Though I have not written Java in 20 years, so I am no expert on the details.)

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java [hunk 1]
```diff
@@ -1094,7 +1095,7 @@ private static String getAscDesc(ModifierSet modifierSet) {
         ascDesc = "DESC";
         break;
       default:
-        // ignore
+        throw new CQLFeatureUnsupportedException("CQL: Unsupported modifier " + modifier.getType());
       }
```
Comment (adamdickmeiss) [inline domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:1099]: Wonder why this is not covered in the tests. It looks as if it should be.

**Review by adamdickmeiss — COMMENTED** [review]
I wonder why the throw

File: domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java (hunk 1 above)
Comment (julianladisch) [reply domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java:1099]: The wrong sortBy column caused the non-optimized code path been executed.

**Review by adamdickmeiss — APPROVED** [review]

**Review by hjiebsco — APPROVED** [review]

**Pull request merged**